packages/flutter_tools/lib/src/test/test_compiler.dart

```diff
@@ -186,7 +186,9 @@ class TestCompiler {
           // The idea is to keep the cache file up-to-date and include as
           // much as possible in an effort to re-use as many packages as
           // possible.
-          globals.fsUtils.ensureDirectoryExists(testFilePath);
+          if (!testCache.parent.existsSync()) {
```

Comment:
This was calling the real fsUtils in the test. Rather than inject it I replaced this with roughly equivalent code. I'm not sure how useful ensureDirectoryExists actually is.